core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -238,30 +238,30 @@ sealed abstract class Parser[+A] {
     * prefer to use methods such as `~`, `*>`, or `<*` when possible,
     * since these are much more efficient.
     */
-  def flatMap[B](fn: A => Parser[B]): Parser[B] =
-    Parser.flatMap(this)(fn)
+  def flatMap[B](fn: A => Parser0[B]): Parser0[B] =
+    Parser.flatMap0(this)(fn)
 
   /** Replaces parsed values with the given value.
     */
-  def as[B](b: B): Parser[B] =
-    Parser.as(this, b)
+  def as[B](b: B): Parser0[B] =
+    Parser.as0(this, b)
 
   /** Wrap this parser in a helper class, enabling better composition
-    * with `Parser1` values.
+    * with `Parser` values.
     *
-    * For example, with `p: Parser[Int]` and `p1: Parser[Double]`:
+    * For example, with `p: Parser0[Int]` and `p1: Parser0[Double]`:
     *
-    *     val a1: Parser[(Int, Double)]  = p ~ p1
-    *     val a2: Parser1[(Int, Double)] = p.with1 ~ p1
+    *     val a1: Parser0[(Int, Double)]  = p ~ p1
+    *     val a2: Parser[(Int, Double)] = p.with1 ~ p1
     *
-    *     val b1: Parser[Double]  = p *> p1
-    *     val b2: Parser1[Double] = p.with1 *> p1
+    *     val b1: Parser0[Double]  = p *> p1
+    *     val b2: Parser[Double] = p.with1 *> p1
     *
-    *     val c1: Parser[Int]  = p <* p1
-    *     val c2: Parser1[Int] = p.with1 <* p1
+    *     val c1: Parser0[Int]  = p <* p1
+    *     val c2: Parser[Int] = p.with1 <* p1
     *
-    * Without using `with1`, these methods will return `Parser` values
-    * since they are not known to return `Parser1` values instead.
+    * Without using `with1`, these methods will return `Parser0` values
+    * since they are not known to return `Parser` values instead.
     */
   def with1: Parser.With1[A] =
```

Comment:
this is maybe the only 1 name left... I guess it is better than withNot0 or something, but still it feels like it sticks out. I guess we should keep it to minimize change.

Reply:
along with `flatMap01` (because there also is flatMap (which was flatMap10) and flatMap0 (which was flatMap), `tailRecM1` which I just didn't see right away and should be changed for consistency, even though it's barely used in user code.